typings/index.d.ts

```diff
@@ -133,17 +134,30 @@ export interface PropGetters<Item> {
 }
 
 export interface Actions<Item> {
-  reset: (otherStateToSet?: Partial<StateChangeOptions<Item>>, cb?: Callback) => void
+  reset: (
+    otherStateToSet?: Partial<StateChangeOptions<Item>>,
```

Comment:
I suppose you do not need `Partial` anymore

Reply:
... only if we make `type` optional.

I'm going to do it now.

Reply:
actually i have my misgivings.

There are times when we know that the library is going to set the `type` property for us. It would be nice not to force people to use `!` (`changes.type!.toUpperCase()`)

Reply:
Not sure what you want to achieve. At the moment you added partial to the interface that is already used with `Partial` wrapper most of the time. It would make sense to remove `Partial` wrapper otherwise this change does not make much sense. Or is it?

Reply:
As of the last commit, `type` is required.

That is not correct most of the time. Most of the time you can pass `StateChangeOptions` without `type`. However I still like having the option of telling the compiler in certain instances, that `type` exists and is not null.

An example would be the first parameter passed to `onStateChange`, or the parameter passed to `stateReducer`. We can be confident that `type` will always be defined on those objects.

It's in these instances that having `type` required is helpful.